### PR TITLE
[CARBONDATA-744] The property "spark.carbon.custom.distribution" should be change to carbon.custom.block.distribution and should be part of CarbonProperties

### PR DESCRIPTION
**Pull request by mohammadshahidkhan**
1. changed property name from spark.carbon.custom.distribution to carbon.custom.distribution
2. Added the property in CarbonProperties
3. Added the test case for custom carbon Block distribution

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1003/

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala [hunk 1]
```diff
@@ -116,8 +116,8 @@ class CarbonScanRDD(
           i += 1
           result.add(partition)
         }
-      } else if (sparkContext.getConf.contains("spark.carbon.custom.distribution") &&
-                 sparkContext.getConf.getBoolean("spark.carbon.custom.distribution", false)) {
+      } else if (java.lang.Boolean
+        .parseBoolean(CarbonProperties.getInstance().getProperty("carbon.custom.distribution"))) {
```
Comment (chenliang613) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala]: The PR's title mentions that the property is "spark.carbon.custom.distribution", but here you change the property name to "carbon.custom.distribution", why ?

Comment (kumarvishal09) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala]: I think carbon.custom.block.distribution is better

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1034/

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala [hunk 2]
```diff
@@ -116,8 +116,8 @@ class CarbonScanRDD(
           i += 1
           result.add(partition)
         }
-      } else if (sparkContext.getConf.contains("spark.carbon.custom.distribution") &&
-                 sparkContext.getConf.getBoolean("spark.carbon.custom.distribution", false)) {
+      } else if (java.lang.Boolean
+        .parseBoolean(CarbonProperties.getInstance().getProperty("carbon.block.distribution"))) {
```
Comment (ravipesala) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala]: Please add CarbonCommonConstants and add default value as well.

Comment (mohammadshahidkhan) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala]: fixed review comments
changed property to carbon.custom.block.distribution
added into CarbonCommonConstants

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala (hunk 1 above)
Comment (mohammadshahidkhan) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala]: @kumarvishal09 agree will rename to carbon.custom.block.distribution

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1104/

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala [hunk 3]
```diff
@@ -116,8 +117,9 @@ class CarbonScanRDD(
           i += 1
           result.add(partition)
         }
-      } else if (sparkContext.getConf.contains("spark.carbon.custom.distribution") &&
-                 sparkContext.getConf.getBoolean("spark.carbon.custom.distribution", false)) {
+      } else if (java.lang.Boolean.parseBoolean(CarbonProperties.getInstance()
+        .getProperty(CarbonCommonConstants.CARBON_CUSTOM_BLOCK_DISTRIBUTION,
```
Comment (ravipesala) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala]: Please simplify as follows
```
CarbonProperties.getInstance()
        .getProperty(CarbonCommonConstants.CARBON_CUSTOM_BLOCK_DISTRIBUTION,
          CarbonCommonConstants.CARBON_CUSTOM_BLOCK_DISTRIBUTION_DEFAULT).toBoolean
```

Comment (mohammadshahidkhan) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala]: Fixed

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1144/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**